src/algo/depth-to-rgb-calibration/k-to-dsm.cpp

```diff
@@ -852,14 +840,14 @@ double3 k_to_DSM::laser_incident_direction(double2 angle_rad)
     return laser_incident_direction;
 }
 
-std::vector<double3> k_to_DSM::transform_to_direction(std::vector<double3> vec)
+std::vector< double3 > k_to_DSM::transform_to_direction( std::vector< double3 > const & vec )
 {
-   std::vector<double3> res(vec.size());
-
-   for (auto i = 0; i < vec.size(); i++)
-   {
-       auto norm = sqrt(vec[i].x*vec[i].x + vec[i].y*vec[i].y + vec[i].z*vec[i].z);
-       res[i] = { vec[i].x / norm, vec[i].y / norm, vec[i].z / norm };
-   }
-   return res;
+    std::vector< double3 > res( vec.size() );
+
+    for( auto i = 0; i < vec.size(); i++ )
+    {
+        auto norm = sqrt( vec[i].x * vec[i].x + vec[i].y * vec[i].y + vec[i].z * vec[i].z );
+        res[i] = { vec[i].x / norm, vec[i].y / norm, vec[i].z / norm };
```

Comment:
The `norm` in theory can be 0 - is there an input check or can it be prevented otherwise?

Reply:
I think @aangerma needs to answer this :)